tensorboard/webapp/hparams/_redux/utils_test.ts

```diff
@@ -0,0 +1,307 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {DiscreteFilter, IntervalFilter} from '../types';
+import {buildDiscreteFilter, buildIntervalFilter} from './testing';
+import {
+  combineDefaultHparamFilters,
+  combineDefaultMetricFilters,
+  getIdFromExperimentIds,
+} from './utils';
+
+describe('hparams/_redux/utils test', () => {
+  describe('#getIdFromExperimentIds', () => {
+    it('forms a unique id from experiment ids', () => {
+      expect(getIdFromExperimentIds(['foo', 'bar'])).toBe('["bar","foo"]');
+    });
+
+    it('sorts experiment ids for consistent id', () => {
+      expect(getIdFromExperimentIds(['foo', '1', 'bar'])).toBe(
+        '["1","bar","foo"]'
+      );
+    });
+  });
+
+  describe('#combineDefaultHparamFilters', () => {
+    it('combines filters with the same name', () => {
+      const result = combineDefaultHparamFilters([
+        new Map([
```

Comment:
In the second case, the type inference thinks that because the first item is `DiscreteFilter`, map is `Map<string, DiscreteFilter>` so any other mismatching type, e.g., `IntervalFilter`, is a wrong one. So, I have to tell Map constructor that it can have two different types of values.